src/main/java/org/gridsuite/directory/server/DirectoryService.java

```diff
@@ -34,29 +33,29 @@ private static ElementAttributes fromEntity(DirectoryElementEntity entity) {
         return new ElementAttributes(entity.getId(), entity.getName(), ElementType.valueOf(entity.getType()), new AccessRightsAttributes(entity.isPrivate()), entity.getOwner());
     }
 
-    public Mono<DirectoryElementEntity> createDirectory(DirectoryAttributes directoryAttributes) {
-        return Mono.fromCallable(() -> directoryElementRepository.save(new DirectoryElementEntity(UUID.randomUUID(), directoryAttributes.getParentId(),
-                                                                   directoryAttributes.getDirectoryName(),
-                                                                   ElementType.DIRECTORY.toString(),
-                                                                   directoryAttributes.getAccessRights() == null || directoryAttributes.getAccessRights().isPrivate(),
-                                                                   directoryAttributes.getOwner())));
+    public Mono<ElementAttributes> createElement(ElementAttributes elementAttributes, UUID directoryUuid) {
+        return Mono.fromCallable(() -> fromEntity(directoryElementRepository.save(new DirectoryElementEntity(
+                elementAttributes.getElementUuid() == null ? UUID.randomUUID() : elementAttributes.getElementUuid(),
+                directoryUuid,
+                elementAttributes.getElementName(),
+                elementAttributes.getType().toString(),
+                elementAttributes.getAccessRights() == null || elementAttributes.getAccessRights().isPrivate(),
+                elementAttributes.getOwner()))));
     }
 
-    public Mono<DirectoryElementEntity> addElementToDirectory(Optional<String> directoryUuid, ElementAttributes elementAttributes) {
-        return Mono.just(directoryElementRepository.save(new DirectoryElementEntity(elementAttributes.getElementUuid(),
-                                                                   directoryUuid.isPresent() ? UUID.fromString(directoryUuid.get()) : null,
-                                                                   elementAttributes.getElementName(),
-                                                                   elementAttributes.getType().toString(),
-                                                                   elementAttributes.getAccessRights().isPrivate(),
-                                                                   elementAttributes.getOwner())));
+    public Mono<ElementAttributes> createElement(RootDirectoryAttributes rootDirectoryAttributes, UUID directoryUuid) {
```

Comment:
Why keep method name createElement, which is generic, instead of naming it createRootDirectory, since the method is designed to create only root directories ?